setup.py

```diff
@@ -12,22 +12,89 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
-import setuptools
 from setuptools import setup
+from setuptools.command.build_ext import build_ext
+from setuptools.extension import Extension
+
 import versioneer
 
 # read the contents of your README file
 from pathlib import Path
 this_directory = Path(__file__).parent
 long_description = (this_directory / "README.md").read_text()
 
+ZITI_SDK_BASE = 'https://github.com/openziti/ziti-sdk-c/releases/download'
+
+
+class GetZitilib(build_ext):
+
+    def build_extension(self, ext) -> None:
+        import platform
+        ziti_ver = self.get_sdk_version()
+        osname, arch, libname = self.get_platform()
+        sdk_distro = self.download_sdk(ziti_ver, osname, arch)
+        self.extract_zitilib(sdk_distro, libname, self.build_lib)
+
+    def get_platform(self):
+        import platform
+        osname = platform.system()
+        mach = platform.machine()
+        arch, _ = platform.architecture()
+        if osname == 'Linux':
+            if mach.startswith('arm'):
+                if arch == '32bit':
+                    mach = 'arm'
+                elif arch == '64bit':
+                    mach = 'arm64'
+            return osname, mach, 'libziti.so'
+        elif osname == 'Darwin':
+            return osname, mach, 'libziti.dylib'
+        elif osname == 'Windows':
+            return osname, mach, 'ziti.dll'
+
+    def get_sdk_version(self):
+        opts = self.distribution.get_option_dict('openziti')
+        _, ver = opts['ziti_sdk_version']
+        return ver
+
+    def extract_zitilib(self, distro, libname, target):
+        import zipfile
+        from io import BytesIO
+        with zipfile.ZipFile(BytesIO(distro)) as zipf:
+            return zipf.extract(member=f'lib/{libname}', path=f'{target}/openziti')
+
+    def download_sdk(self, version, osname, arch):
+        from urllib.error import HTTPError
+        from urllib.request import Request, urlopen
+        filename = f'{ZITI_SDK_BASE}/{version}/ziti-sdk-{version}-{osname}-{arch}.zip'
+        headers = {}
+        req = Request(url=filename, headers=headers)
+        try:
+            with urlopen(req) as response:
+                length = response.getheader('content-length')
+                if response.status != 200:
+                    raise Exception(f'Could not download "{filename}"')
+                    return None
+
+                print(f"Downloading {length} from {filename}")
+                return response.read()
+        except HTTPError:
+            raise
```

Comment:
This doesn't do anything useful here. Catching an exception only to raise it is the same as not catching it at all. You only want to do this when you want to say, add some functionality, like "Log this somewhere and then raise"